### ISSUE TITLE
Can't set upgrade authority from Ledger due to "Ledger operation not supported"

### ISSUE THREAD

**Issue by 0xrooters** [solana-labs/solana#19157]
#### Problem
I'm having trouble using `solana program set-upgrade-authority` with a Ledger.

I deployed a program on devnet, then set the authority to my Ledger address with:

`solana program set-upgrade-authority <program-address> --new-upgrade-authority usb://ledger`

To test that it's fully functional, I then tried to change it back:

`solana program set-upgrade-authority <program-address> --upgrade-authority usb://ledger --new-upgrade-authority <address>`

which fails with:
```
Waiting for your approval on Ledger hardware wallet usb://ledger/<address>
Error: Ledger operation not supported
```

Other functionality works with the Ledger, e.g. `solana transfer <address> 0 --keypair usb://ledger` worked fine.

The error message seems to come from [here](https://github.com/solana-labs/solana/blob/52f4b96a80e2159847d6bd749ce12b126dedb240/remote-wallet/src/ledger_error.rs#L29), so seems `set-upgrade-authority` is trying to use an unsupported SDK (not sure what that means).

Basically the impact of this is that we can't use a Ledger to secure the upgrade authority for now.

**Comment by CriesofCarrots** [comment]
The [solana ledger app](https://github.com/LedgerHQ/app-solana) is only able to support parsing a limited, explicit set of instruction types, and `set-upgrade-authority` is not currently one of these. I can see the use-case for this one; we can look at adding support in the future.
In the meantime, the way to sign these transactions via the CLI with a ledger is by turning on blind signing in the app.

**Comment by 0xrooters** [comment]
That workaround worked! Thanks
